### PR TITLE
Add more tests for FileReader.result.

### PR DESCRIPTION
**Pull request by mkruisselbrink**
In particular this adds tests to verify the value of the attribute
during progress and loadstart events.

I think this matches what the spec currently says, but there are some open questions in https://github.com/w3c/FileAPI/issues/79 about what the spec should actually say, so we should probably not merge this until we figure out what the desired behavior actually is.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18875)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/280143177)

File: FileAPI/reading-data-section/filereader_result.html [hunk 1]
```diff
@@ -54,6 +55,61 @@
...
     }, "readAsArrayBuffer");
+
+    async_test(function() {
+      var readBinaryString = new FileReader();
+      assert_equals(readBinaryString.result, null);
+
+      readBinaryString.onloadend = this.step_func(function(evt) {
+        assert_equals(typeof readBinaryString.result, "string", "The result type is string");
+        assert_equals(readBinaryString.result, "This test the result attribute", "The result is correct");
+        console.log(readBinaryString.result);
```
Comment (inexorabletash) [inline FileAPI/reading-data-section/filereader_result.html]: Remove the log statement

File: FileAPI/reading-data-section/filereader_result.html [hunk 2]
```diff
@@ -54,6 +55,61 @@
...
+        reader.addEventListener(event, this.step_func_done(function(evt) {
+          assert_equals(reader.result, null);
+        }));
+
+        reader[readMethod](blob);
+      }, 'result is null during "' + event + '" event for ' + readMethod);
+
+      async_test(function() {
+        var reader = new FileReader();
+        reader.onloadend = this.step_func(function(evt) {
```
Comment (inexorabletash) [inline FileAPI/reading-data-section/filereader_result.html]: Should this null out `reader.onloadend` so it isn't trying to run this code a second time (after done, in theory...)

**Review by inexorabletash — COMMENTED** [review]
Sorry about the delay - is this still valid?

File: FileAPI/reading-data-section/filereader_result.html (hunk 2 above)
Comment (mkruisselbrink) [reply FileAPI/reading-data-section/filereader_result.html]: Changed to promise_test using EventWatcher, so no longer applicable.

**Comment by mkruisselbrink** [comment]
> Sorry about the delay - is this still valid?

Sorry about the delay, yes this is still valid. These are tests for the spec changes in https://github.com/w3c/FileAPI/pull/118/

File: FileAPI/reading-data-section/filereader_result.html [hunk 3]
```diff
@@ -54,6 +55,43 @@
...
+          var reader = new FileReader();
+          assert_equals(reader.result, null, 'result is null before read');
+
+          var eventWatcher = new EventWatcher(t, reader,
+              [event, 'loadend']);
+
+          reader[method](blob);
+          assert_equals(reader.result, null, 'result is null after first read call');
+          await eventWatcher.wait_for(event);
+          assert_equals(reader.result, null, 'result is null during event');
```
Comment (inexorabletash) [inline FileAPI/reading-data-section/filereader_result.html]: Nice... and all browsers have converged on including the microtask execution within the event dispatch.

**Review by inexorabletash — APPROVED** [review]
LGTM!

**Pull request merged**